incubator/kafka/templates/statefulset.yaml

```diff
@@ -28,8 +28,10 @@ spec:
       - name: init-ext
         image: "{{ .Values.external.init.image }}:{{ .Values.external.init.imageTag }}"
         imagePullPolicy: "{{ .Values.external.init.imagePullPolicy }}"
-        args:
-        - -n ${POD_NAMESPACE} label pods ${POD_NAME} pod=${POD_NAME}
+        command:
```

Comment:
@josdotso can you go into more detail about why we need to `sh` vs just use the scratch image? I don't follow the problem totally.

Reply:
@benjigoldberg 

Using a from scratch image did not label the pods at all, exited 0, and produced no init container log at all. 

With `sh` in the mix, we can `set -euxc` which tells `sh` to error upon any non-zero command, error on any vars it cannot interpolate, enable shell verbosity enough to tell us what shell command ran.  stdout gives useful output anyway -- but only on success. Even if the scratch approach worked in most cases, it would never tell us what shell command started kubectl.  Therefore, we must use a shell here to properly debug the interpolation of the variables.

When label is not present, the services created for external access are set to use dead endpoints. This happens because selector doesn't match unlabeled pods.

Here's an example log output after this PR:

```bash
$ kubectl logs kafka-kafka-0 -c init-ext
+ kubectl label pods kafka-kafka-0 --namespace default pod=kafka-kafka-0
pod "kafka-kafka-0" labeled
```

Reply:
Thanks for the explanation 👍 makes sense to me